cow_amm/profitability/invariant_growth/cow_amm_4059213.sql

```diff
@@ -0,0 +1,92 @@
+-- Computes the absolute surplus per $100 tvl per day (aka its invariant growth) for a CoW AMM.
+-- Parameters
+--  {{token_a}} - either token of the desired uni pool
+--  {{token_b}} - other token of the desired uni pool
+--  {{start}} - date as of which the analysis should run
+
+-- Finds the CoW AMM pool address given tokens specified in query parameters (regardless of order) 
+with cow_amm_pool as (
+    select
+        block_time,
+        address,
+        token_1_address,
+        token_2_address
+    from query_3959044
+    where ((token_1_address = {{token_a}} and token_2_address = {{token_b}}) or (token_2_address = {{token_a}} and token_1_address = {{token_b}}))
+    order by 1 desc
+    limit 1
+),
+
+-- Computes the token balance changes of the relevant token per transaction
+reserves_delta as (
```

Comment:
Most of the query can be substitutes by calling the base query.

The base query also tries to take care of
* liquidity injections
* deposits and withdrawals

If it is actually this easy, the simplifications should be added to the base query as well.

Reply:
Which base query do you mean exactly? I created #27 (but am not aware of an existing one)

Reply:
I meant [query 3968186](https://dune.com/queries/3968186), which is currently used for dashboards. It is not currently part of this repo.